changelog/_unreleased/2025-10-14-support-all-robots-txt-directives.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Support all robots.txt directives including User-agent blocks
+issue: NEXT-12787
+author: Björn Meyer
+author_email: b.meyer@shopware.com
+author_github: BrocksiNet
+---
+# Core
+* Added `Shopware\Storefront\Page\Robots\Parser\RobotsDirectiveParser` to parse robots.txt configurations
+* Added `Shopware\Storefront\Page\Robots\Parser\ParsedRobots` value object for parsed robots.txt data
+* Added `Shopware\Storefront\Page\Robots\Struct\RobotsDirective` value object representing a single directive
+* Added `Shopware\Storefront\Page\Robots\Struct\RobotsUserAgentBlock` value object representing a user-agent block
+* Added `globalUserAgentBlocks` property to `Shopware\Storefront\Page\Robots\RobotsPage` to store global user-agent blocks
+* Changed `Shopware\Storefront\Page\Robots\RobotsPageLoader` to parse and separate global user-agent blocks from domain-specific path rules
+* Changed `Shopware\Storefront\Page\Robots\Struct\DomainRuleStruct` to support new parser while maintaining backward compatibility
+* Added new system configuration option `core.basicInformation.robotsDisableDefaults` to disable Shopware's default robots.txt rules
+
+___
+# Storefront
+* Changed rendering of robots.txt to support custom User-agent blocks
+* Changed robots.txt template to respect new `robotsDisableDefaults` configuration
+
+___
+# Upgrade Information
+
+## Technical Changes (Backward Compatible)
```

Comment:
remember that this will be the headline in the upgrade file. 

`Technical Changes (Backward Compatible)` will be quite lost without its context. 

so probably, you should add a title for the whole topic and all `##` headlines should become `###` headlines